### PR TITLE
Overview Tab - Layers

### PR DESCRIPTION
**Pull request by jws-1**
This small PR adds `ModuleLayers` to the `OverviewTab`, as well as displaying (but not offering means to edit) their corresponding `Modules`. Future work (see #1652) will make this view more aesthetically pleasing (for example, ensuring that `Nodes` are sized to fit their contents), as well as providing means to edit `ModuleLayers` and control their `Modules`. Furthermore, the `ModuleListView` currently cannot be navigated, but this will be dealt with in a future PR along with the previously described visual improvements.

### CODE REVIEW

File: src/gui/models/dissolveModelImageProvider.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include "gui/models/dissolveModel.h"
+#include "gui/models/moduleLayerModel.h"
+
+DissolveModelImageProvider::DissolveModelImageProvider(DissolveModel *dissolveModel)
+    : QQuickImageProvider(QQmlImageProviderBase::Image)
+{
+    dissolveModel_ = dissolveModel;
+}
+
+QImage DissolveModelImageProvider::requestImage(const QString &id, QSize *size, const QSize &requestedSize)
```
Comment (trisyoungs) [inline src/gui/models/dissolveModelImageProvider.cpp:14]: Hmm. Perhaps a contender for my new most-despised Qt override!

File: src/gui/models/moduleLayersModel.cpp [hunk 2]
```diff
@@ -2,34 +2,61 @@
...
     endResetModel();
 }
 
-ModuleLayer *ModuleLayersModel::rawData(const QModelIndex &index) const { return layers_->get()[index.row()].get(); }
+ModuleLayer *ModuleLayersModel::rawData(const QModelIndex &index) const
+{
+    if (!layers_)
+        return {};
+    if (!index.isValid())
+        return {};
```
Comment (trisyoungs) [inline src/gui/models/moduleLayersModel.cpp]:
```suggestion
    if (!layers_ || !index.isValid())
        return {};
```

File: src/gui/models/moduleLayersModel.cpp [hunk 3]
```diff
@@ -2,34 +2,61 @@
...
+
     switch (role)
     {
         case (Qt::DisplayRole):
-            return QString::fromStdString(std::string(rawData(index)->name()));
+            return QString::fromStdString(std::string(data->name()));
         case (Qt::UserRole):
-            return QVariant::fromValue(rawData(index));
+            return QVariant::fromValue(data);
+        case (Qt::UserRole + 1):
```
Comment (trisyoungs) [inline src/gui/models/moduleLayersModel.cpp]: I think we definitely need a custom `enum` using `Qt::UserRole` as the base here.

**Review by trisyoungs — APPROVED** [review]
Apart from the ugliness this is looking pretty good!

File: src/gui/models/moduleLayersModel.cpp (hunk 3 above)
Comment (jws-1) [reply src/gui/models/moduleLayersModel.cpp]: Ah yep, I meant to do that but forgot to come back!

**Pull request merged**